web/src/games/bingo/components/playCard.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import { blue, red } from '@material-ui/core/colors';
+
+import { INumberState } from '../definitions';
+import { GRID_SIZE, CALL_BOX_SIZE, STROKE_WIDTH, WILDCARD_NUM } from '../constants';
+import commonCSS from './biComponent.css';
+
+const fbgIcon = require('../media/fbg_icon.png');
+const cornerAdjustment = 1 + 1 / GRID_SIZE;
+
+interface IOneCardProps {
+  number: INumberState;
+  onNumberClicked: (number: INumberState) => void;
+}
+
+function OneCard(props: IOneCardProps) {
+  const { number } = props;
+  const xPos = Math.floor(number.id / GRID_SIZE);
+  const yPos = number.id % GRID_SIZE;
+  const boxMargin = STROKE_WIDTH;
+  const imageScale = 0.2;
+  const isWildcard = number.value === WILDCARD_NUM;
+
+  return (
+    <g
+      key={`bi_card_group_${number.id}`}
+      data-testid={`bi-card-group-${number.id}`}
+      onClick={() => {
+        if (!isWildcard) {
+          props.onNumberClicked(number);
+        }
+      }}
+    >
+      <rect
```

Comment:
This is a very large function and hard to read. Can you break down sub-components (i.e. the rect) into its own private method?

Reply:
No No. The rect and the text must be in sync. I think this can stay. They use common xPos and yPos, etc.